test/extensions/filters/http/lua/lua_filter_test.cc

```diff
@@ -841,7 +842,7 @@ TEST_F(LuaHttpFilterTest, HttpCall) {
   response_message->body().add(response, 8);
   EXPECT_CALL(*filter_, scriptLog(spdlog::level::trace, StrEq(":status 200")));
   EXPECT_CALL(*filter_, scriptLog(spdlog::level::trace, StrEq("8")));
-  EXPECT_CALL(*filter_, scriptLog(spdlog::level::trace, StrEq("resp")));
+  EXPECT_CALL(*filter_, scriptLog(spdlog::level::trace, StrEq("resp0nse")));
```

Comment:
Sorry, why we do have this additional `0nse` again?

Reply:
At pervious test, `request_hanle:logTrace(body)` want to print "resp\0nse". However, testing:StrEq can not handling this situation with string contains '\0'. So I change `request_handle:logTrace(body)` to  `request_handle:logTrace(string.sub(body, 1, 4) .. string.byte(body, 5) .. string.sub(body, 6, 8))`.

Reply:
It would be better to just keep only one line log body in the lua script.

```
request_handle:logTrace(body)
```

And use only one `EXPECT_CALL` to validate it. Remote all others unnecessary `EXPECT_CALL`.

```
EXPECT_CALL(*filter_, scriptLog(spdlog::level::trace, StrEq(std::string(response, 8))));
```

But I have no particular insistence on this.